### ISSUE TITLE
proposal to change as_html() return value in scalar context

### ISSUE THREAD

**Issue by mla** [tokuhirom/Web-Query#52]
I realize it's documented, but I was tripped up by the fact that as_html() in a scalar context returns the string representation of only the *first* element.

I was doing something like:

```
  my $html = $dom->as_string;
```

And couldn't figure out why it was getting truncated until I traced it back to this method.

I think it's more intuitive to return the concatenation of all the stringified values in this context. If you only want to stringify the first element (which seems like a rarer case), you can filter out just the first element.

Since the API is listed as alpha and I kind of doubt many people depend on stringifying just the first element, I thought it might be an acceptable change.

**Comment by mla** [comment]
Yes, good point @yanick. And I just checked and jquery itself behaves that way too. So my suggestion is right out.

I'm wondering if I should just submit a doc change to elaborate a bit more on how as_html() behaves with a list of elements.

**Comment by yanick** [comment]
> I'm wondering if I should just submit a doc change to elaborate a bit more on how as_html() behaves with a list of elements.

That'd be fantastic. Meanwhile, I'll try to see if my solution above doesn't break anything.
